### PR TITLE
 Merge branch OpenXRay/xd_dev and fix conflicts

### PR DESCRIPTION
**Pull request by Zegeri**
Most of the conflicts arise from the xrUICore migration and the case-sensitive headers fixes.

### CODE REVIEW

**Comment by eagleivg** [comment]
xrUICore building with errors...

**Comment by q4a** [comment]
> xrUICore building with errors...

I'm not sure. Travis build it without errors: https://travis-ci.org/q4a/xray-16/jobs/435485672

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-X-Ray Engine 1.6 expansion
+X-Ray Engine 1.6 expansion [![Codacy Badge](https://api.codacy.com/project/badge/Grade/40d84a83b38147c49b4e66705c47a16a)](https://www.codacy.com/app/OpenXRay/xray-16?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=OpenXRay/xray-16&amp;utm_campaign=Badge_Grade)
```
Comment (Zegeri) [inline README.md]: Should this point to this fork instead?

Comment (Xottab-DUTY) [reply README.md]: If Codacy is connected to it, then yes.

Comment (q4a) [reply README.md]: No, linux branch will not use Codacy. Codacy link should poit to OpenXRay

**Comment by Zegeri** [comment]
Looks like Clang can't build this right now.

**Comment by Zegeri** [comment]
I am going to add a patch by @eagleivg  to fix this issue, but as commented by him: 
>I found two more problems: the resolution is not saved and the transition to windowed mode does not work in directx-based renders

I can't reproduce or fix this issues in my current setup, so I won't be adding commits of my own any time soon.

**Comment by Zegeri** [comment]
I have resolved the new conflicts in the latest commits. Notice that there have been commits in xd_dev in the last 5 days, but I'm not including them here.

**Pull request merged**